### PR TITLE
Kafka source oauth new

### PR DESCRIPTION
**Pull request by ajeeshakd**
### Description
- OAuth functionality implemented for the Kafka Source Consumer
 
### Issues Resolved
- Github issue https://github.com/opensearch-project/data-prepper/issues/254
 
### Check List
- [ ] OAuth functionality includes testing. 
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kkondaka** [comment]
Please add tests for all these files.

File: data-prepper-plugins/kafka-plugins/src/test/resources/sample-pipelines.yaml [hunk 1]
```diff
@@ -25,10 +24,25 @@ log-pipeline:
...
+        - name: my-topic-2
           group_id: DPKafkaProj-1
       schema:
         registry_url: http://localhost:8081/
         version: 1
+      authentication:
+        sasl_oauth:
+          oauth_client_id: 0oa9wc21447Pc5vsV5d7
+          oauth_client_secret: aGmOfHqIEvBJGDxXAOOcatiE9PvsPgoEePx8IPPa
+          oauth_login_server: https://dev-13650048.okta.com
```
Comment (ashoktelukuntla) [inline data-prepper-plugins/kafka-plugins/src/test/resources/sample-pipelines.yaml:36]: As discussed - Most of these should be optional. Did you check in the code after making changes ?

Comment (ajeeshakd) [reply data-prepper-plugins/kafka-plugins/src/test/resources/sample-pipelines.yaml:36]: Yes, we did it, as per the latest code commit, the following configs are mandatory and rest of them are optional.

1.  oauth_client_id
2.  oauth_client_secret
3.  oauth_login_server
4.  oauth_login_endpoint

**Comment by kkondaka** [comment]
@ajeeshakd ,
I think we need the auth config something like this.  And OAUTH config goes under `oauth:` and these changes can go there.
```
authentication:
  protocol: # plaintext/ssl (more may be added later)
     plaintext:
     ssl:
  mechanism: # plain/iam/oauth (more like - scram, gssapi may be added later)
     plain:
     iam:
     oauth:
```
Let me know if this makes sense.

**Comment by ajeeshakd** [comment]
Closing this PR because the review comments change has been done as part of the PR https://github.com/opensearch-project/data-prepper/pull/2935

**Pull request closed without merging**